src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -358,6 +358,25 @@ public function directive($name, callable $handler)
     }
 
     /**
+     * Register a custom directive representing an if statement
+     * 
+     * @param  string   $name
+     * @param  callable $condition
+     * @return void
+     */
+    public function if($name, callable $condition)
+    {
+        $this->directive($name, function (...$args) use ($condition) {
+            $condition = (int) (true === $condition(...$args));
```

Comment:
And why are your comparing to `true` ? You could just return `<?php if ($condition(...args): ?>`, no ?